mythril/analysis/modules/state_change_external_calls.py

```diff
@@ -64,137 +84,82 @@ def __init__(self):
+                constraints + [UGT(gas, symbol_factory.BitVecVal(2300, 256)), to > 16]
             )
 
             # Check whether we can also set the callee address
             try:
                 constraints += [to == 0xDEADBEEFDEADBEEFDEADBEEFDEADBEEFDEADBEEF]
                 solver.get_model(constraints)
-                annotations[0].calls.append(
-                    CallIssue(call=call, user_defined_address=True)
-                )
-            except UnsatError:
-                annotations[0].calls.append(
-                    CallIssue(call=call, user_defined_address=False)
-                )
 
+                global_state.annotate(StateChangeCallsAnnotation(global_state, True))
+            except UnsatError:
+                global_state.annotate(StateChangeCallsAnnotation(global_state, False))
         except UnsatError:
             pass
 
     @staticmethod
-    def _analyze_state(state: GlobalState) -> List[Issue]:
-        """
+    def _analyze_state(global_state: GlobalState) -> List[Issue]:
 
-        :param state:
-        :return:
-        """
-        address = state.get_current_instruction()["address"]
         annotations = cast(
             List[StateChangeCallsAnnotation],
-            list(state.get_annotations(StateChangeCallsAnnotation)),
+            list(global_state.get_annotations(StateChangeCallsAnnotation)),
         )
-        opcode = state.get_current_instruction()["opcode"]
+        op_code = global_state.get_current_instruction()["opcode"]
 
         if len(annotations) == 0:
-            if opcode in ("SSTORE", "CREATE", "CREATE2"):
+            if op_code in ("SSTORE", "CREATE", "CREATE2"):
                 return []
-            log.debug("Creating annotation for state")
-            state.annotate(StateChangeCallsAnnotation())
-            annotations = cast(
-                List[StateChangeCallsAnnotation],
-                list(state.get_annotations(StateChangeCallsAnnotation)),
-            )
-
-        if opcode in ("SSTORE", "CREATE", "CREATE2"):
-            return StateChange._handle_state_change(
-                state, address=address, annotation=annotations[0]
-            )
-        call = get_call_from_state(state)
-
-        if call is None:
-            return []
-
-        if StateChange._balance_change(call.value):
-            return StateChange._handle_state_change(
-                state, address=address, annotation=annotations[0]
-            )
-
-        if opcode == "CALL":
-            StateChange._add_external_call(state, annotations=annotations)
-
-        return []
+        if op_code in ("SSTORE", "CREATE", "CREATE2"):
+            for annotation in annotations:
+                annotation.state_change_states.append(global_state)
+
+        # Record state changes following from a transfer of ether
+        if op_code in ("CALL", "DELEGATECALL", "CALLCODE"):
+            value: BitVec = global_state.mstate.stack[-3]
```

Comment:
This type hinting isn't supported in py35

Reply:
😢